presto-main/src/main/java/com/facebook/presto/sql/MaterializedViewUtils.java

```diff
@@ -0,0 +1,88 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.facebook.presto.sql;
+
+import com.facebook.presto.spi.SchemaTableName;
+import com.facebook.presto.sql.analyzer.Analysis;
+import com.facebook.presto.sql.analyzer.MaterializedViewPlanValidator;
+import com.facebook.presto.sql.tree.Node;
+import com.facebook.presto.sql.tree.Query;
+import com.facebook.presto.sql.tree.Statement;
+import com.google.common.collect.ImmutableMap;
+
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+
+import static com.facebook.presto.metadata.MetadataUtil.toSchemaTableName;
+import static com.facebook.presto.spi.ConnectorMaterializedViewDefinition.TableColumn;
+import static com.facebook.presto.sql.analyzer.MaterializedViewPlanValidator.MaterializedViewPlanValidatorContext;
+import static com.google.common.collect.ImmutableMap.toImmutableMap;
+
+public class MaterializedViewUtils
+{
+    private MaterializedViewUtils() {}
+
+    public static void validateMaterializedViewQueryPlan(Statement query)
+    {
+        MaterializedViewPlanValidator validator = new MaterializedViewPlanValidator(query);
+        validator.process(query, new MaterializedViewPlanValidatorContext());
```

Comment:
This is a weird interface. We are using `query` in the construct and also pass it through a function. There must be some duplication here.

Reply:
good catch

Reply:
It seems we indeed are passing through contructor?

Reply:
@gggrace14 are you planning to address this comment?

Reply:
Not sure if I understand correctly what you mean here. The constructor we were looking at is the visitor constructor, and what James objected to is we were passing the `viewQuery` to both the visitor constructor and the process() method.

After revision, I'm only passing the `viewQuery` to the process() here, https://github.com/prestodb/presto/blob/99539b6cb7ac5ac2b7701da09c948aa7113cb008/presto-main/src/main/java/com/facebook/presto/sql/analyzer/MaterializedViewPlanValidator.java#L52